### DISCUSSION TITLE
Add bookmark support to forge buffers

### DISCUSSION THREAD

**Discussion by TauPan** [magit/forge#555] [Ideas]
I would appreciate being able to bookmark a forge buffer via burly, see  https://github.com/alphapapa/burly.el/issues/57#issuecomment-1508851512 .

It's possible with orgit to recreate a forge buffer but it's not possible to bookmark it.

This request might not be in the right place, since it would also be good if it was possible to bookmark any kind of magit buffers but I haven't really checked yet if that's already possible. It certainly isn't with forge buffers.

 Integration of bookmark support into all magit-related libraries might be a larger undertaking. I don't know.

**Comment by tarsius** [comment]
See https://github.com/magit/orgit-forge.

**Reply by TauPan** [reply]
That doesn't have support for emacs bookmarks, if I read the code correctly. e.g. there's no local `bookmark-make-record-function`.

Opening links to forge buffers from org-mode already works for me with orgit-forge.

If I try `bookmark-set` in a forge buffer I get the message:

~~~
user-error: Bookmarking is not implemented for forge-topic-mode buffers
~~~

So that would be a new feature, correct?

**Comment by tarsius** [comment]
I've added support for bookmarking topic buffers, in aaeec51cca677e678032c87c26d6024efd7c71c4.

You'll also have to update Magit.

**Reply by TauPan** [reply]
I just upgraded and it works for issues if I make sure that the forge package is loaded before I open the bookmark. I guess I need to update my autoload configuration.

For the issues list from forge I get this error, unless I create the buffer first:

~~~
Burly was unable to get a buffer named: *forge-topics: pet/SIS/sis*
URL: emacs+burly+name://?*forge-topics: pet/SIS/sis*
Please report this error to the developer
~~~

GNU Emacs 29.4 (build 1, x86_64-suse-linux-gnu, GTK+ Version 3.24.43, cairo version 1.18.0)

Current package versions:

- forge-20250605.1548
- orgit-20250601.1057
- orgit-forge-20250601.1059
- magit-20250605.1750

**Reply by TauPan** [reply]
Ah, maybe the issues list is not meant to be persistent... I pressed return on it and got a menu which I cancelled with Ctrl-g.

**Reply by TauPan** [reply]
So it works with the magit buffer itself, a commit, an issue and a PR. Perfect!

Thank you so much! <3 :-D
